mlir/lib/Dialect/Vector/IR/ScalableValueBoundsConstraintSet.cpp

```diff
@@ -62,6 +60,10 @@ ScalableValueBoundsConstraintSet::computeScalableBound(
   int64_t pos = scalableCstr.insert(value, dim, /*isSymbol=*/false);
   scalableCstr.processWorklist();
 
+  // Check the resulting constraints set is valid.
+  if (scalableCstr.cstr.isEmpty())
```

Comment:
I think `.isEmpty()` does the job right now (I was mainly going off the existing value bounds usage). I think this would be fine to replace with `.isEmptyByFMTest()` once your PR lands.